accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,209 @@
+
+Uyuni always gets all products meta information from SCC: which products exists and all repositories assigned.
+These products are only showed in the products setup wizard page if an authentication mechanism for the repositories is available, meaning user have access to it.
+
+The proposed solution will implement a new authentication mechanism on uyuni (named `cloudrmt`, for example) to deal with cloud RMT server authentication. With this approach we will be able to reuse the existing product/channel management features.
+
+### Implementation steps - Cloud RMT server access
+
+To access the cloud RMT server uyuni server needs to know is IP address (which is not registered in DNS) and trust the server certificate.
+
+Cloud RMT IP needs to be added to `/ets/hosts` on uyuni server. IP and hostname are returned by the data extraction script.
+
+Cloud RMT https certificate is also returned by the data extraction tool and on uyuni server we need to:
+  - add the certificate to folder `/etc/pki/trust/anchors/<label>`
+  - run command `update-ca-certificates`
+
+### Implementation steps - Register new PAYG repositories
+
+After we get all repositories information from the PAYG instance, for each repository we will need to:
+
+#### Find internal suse Scc repository ID
+
+Repositories are identified by its URL. We can extract repository endpoint from PAYG machine and find the corresponding repository in Uyuni database. Note that the cloud RMT server have a url prefix `/repo/` which we need to remove before compare.
+
+Example:
+- URL on PAYG instance: `https:\\host/repo/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+- URL uyuni database (suseSccRepository table): `https://updates.suse.com/SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/`
+
+Query to be executed: `select id from susesccrepository where url like '%SUSE/Products/SLE-Module-Basesystem/15-SP2/x86_64/product/';`
+
+The query above will return the internal suse scc repository ID, to be used in next step.
+
+#### Insert repository authentication data
+
+- If not exists, add a new entry to table `susecredentials`, with the new authentication type `cloudrmt`, containing the repository basic authentication credentials.
+- Add a new entry to table `susesccrepositoryauth` for each repository, where:
+  - repo_id: the one obtained in previous step
+  - credentials_id: the one obtain from `susecredentials`
+  - source_id: null
+  - auth_type: `cloudrmt`
+  - auth: authentication header obtained from the PAYG instance for the repository
+
+We also need to save the repository URL, since we will not be able to use the existing SCC repository url.
+If this information will be saved in a new table field of `susesccrepositoryauth` table or in a new table is a implementation detail.
```

Comment:
Check also for suseCredentials table. It has already a "URL" field which could be reused as well. But it also has another meaning. Maybe a good time to clean this up as well.

Reply:
NOw I'm saving the base URL on susecredentials table, and concatenate that with suse scc repository url path, to create the repository URL to be used by reposync.